### DISCUSSION TITLE
Thousands of messages delayed up to an year

### DISCUSSION THREAD

**Discussion by NelsonFrancisco** [rabbitmq/rabbitmq-delayed-message-exchange#221] [Q&A] [answered]
Hello

I'm building a system that requires me to schedule some messages.

The idea is that a data analysis process will run through a dataset and create hundreds, if not thousands of messages, to be sent in the near future (a few minutes) or in the far future (months or a year)

Is this feasible with this plugin? Or should I find another solution?

I've already tested some messages with some minutes, and it's working just fine, but as you might understand, testing for a big period of time is really complicated.

Thank you

**Comment by michaelklishin** [comment, marked as answer]
[Intended Use Cases](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange#intended-use-cases) very explicitly says that this is not a data store, and delaying messages for one year is not one of the design goals (both of the current design and v2 #229) of this plugin.
